src/logging.js

```diff
@@ -73,69 +73,76 @@ function getTransports(isDev) {
     return transports;
 }
 
-type Loggers = {
-    default: Logger,
-    middleware: {
-        requestLogger: Middleware,
-        errorLogger: Middleware,
-    },
-};
-
-function initLogging(logLevel: LogLevel, isDev: boolean): Loggers {
+function initLogging(logLevel: LogLevel, isDev: boolean): Logger {
     // This is the logger that we use to log general information in our app.
     // Whereever one might use console, use this instead.
     const winstonLogger = winston.createLogger<NpmLogLevels>({
         level: logLevel,
         transports: getTransports(isDev),
     });
 
-    // This is the logger that captures requests handled by our express server.
-    const requestLogger = expressWinston.logger({
-        /**
-         * Specify the level that this logger logs at.
-         * (use a function to dynamically change level based on req and res)
-         *     `function(req, res) { return String; }`
-         */
-        level: "info",
-
-        /**
-         * Use the logger we already set up.
-         */
-        winstonInstance: winstonLogger,
-        expressFormat: true,
-        colorize: isDev,
-        meta: false,
-    });
-
-    // This is the logger that captures errors in our express server.
-    const errorLogger = expressWinston.errorLogger({
-        /**
-         * Specify the level that this logger logs at.
-         * (use a function to dynamically change level based on req, res and
-         * err)
-         *     `function(req, res, err) { return String; }`
-         */
-        level: "error",
-
-        /**
-         * Use the logger we already set up.
-         */
-        winstonInstance: winstonLogger,
-    });
-
     winstonLogger.debug(
         `Intialized logging with Level=${logLevel} DeveloperMode=${
             isDev ? "true" : "false"
         }`,
     );
 
-    return {
-        default: winstonLogger,
-        middleware: {requestLogger, errorLogger},
-    };
+    return winstonLogger;
+}
+
+export function extractErrorInfo(error: any): string {
+    if (typeof error === "string") {
+        return error;
+    }
+
+    if (error.response && error.response.error) {
+        return `${error.response.error}: ${error.stack}`;
+    }
+
+    if (error.error && error !== error.error) {
+        return extractErrorInfo(error.error);
+    }
+
+    if (error.stack) {
+        return error.stack;
+    }
+
+    return `${error}`;
+}
+
+export function makeErrorMiddleware(logger: Logger): Promise<Middleware> {
+    // This is the logger that captures errors in our express server.
+    return Promise.resolve(
```

Comment:
Why do you have this return a promise? It doesn't seem like it's ever being used in an async manner so just returning the value should be ok (and easier to understand)!

Reply:
I was aiming for API symmetry and it did need a promise originally. I can change it